File: src/backend/internal/domain/project/phase7.go
package project

import "time"

type SecuritySeverity string

type FindingStatus string

const (
	SecuritySeverityCritical SecuritySeverity = "CRITICAL"
	SecuritySeverityHigh     SecuritySeverity = "HIGH"
	SecuritySeverityMedium   SecuritySeverity = "MEDIUM"
	SecuritySeverityLow      SecuritySeverity = "LOW"
)

const (
	FindingStatusOpen     FindingStatus = "OPEN"
	FindingStatusFixed    FindingStatus = "FIXED"
	FindingStatusAccepted FindingStatus = "ACCEPTED"
)

type SecurityFinding struct {
	ID          string           `json:"id"`
	Title       string           `json:"title"`
	Category    string           `json:"category"`
	Severity    SecuritySeverity `json:"severity"`
	CVSS        float64          `json:"cvss"`
	CVE         string           `json:"cve,omitempty"`
	Description string           `json:"description"`
	PoC         string           `json:"poc,omitempty"`
	File        string           `json:"file,omitempty"`
	Line        int              `json:"line,omitempty"`
	Remediation string           `json:"remediation,omitempty"`
	DetectedBy  string           `json:"detected_by"`
	Status      FindingStatus    `json:"status"`
}

type SecurityAuditSummary struct {
	Score         int `json:"score"`
	CriticalCount int `json:"critical_count"`
	HighCount     int `json:"high_count"`
	MediumCount   int `json:"medium_count"`
	LowCount      int `json:"low_count"`
	TotalFindings int `json:"total_findings"`
}

type SecurityAutoRejectionResult struct {
	Triggered      bool     `json:"triggered"`
	Reason         string   `json:"reason,omitempty"`
	RetryCount     int      `json:"retry_count"`
	ReturnedPhase5 bool     `json:"returned_phase5"`
	Findings       []string `json:"findings,omitempty"`
}

type SecurityAuditReport struct {
	GeneratedAt    time.Time                   `json:"generated_at"`
	Summary        SecurityAuditSummary        `json:"summary"`
	Status         string                      `json:"status"`
	Findings       []SecurityFinding           `json:"findings"`
	Dependencies   []SecurityFinding           `json:"dependencies,omitempty"`
	StaticAnalysis []SecurityFinding           `json:"static_analysis,omitempty"`
	AutoRejection  SecurityAutoRejectionResult `json:"auto_rejection"`
}

File: src/frontend/src/types/phase7.ts
export type SecuritySeverity = "CRITICAL" | "HIGH" | "MEDIUM" | "LOW";
export type SecurityFindingStatus = "OPEN" | "FIXED" | "ACCEPTED";

export interface SecurityFinding {
  id: string;
  title: string;
  category: string;
  severity: SecuritySeverity;
  cvss: number;
  cve?: string;
  description: string;
  poc?: string;
  file?: string;
  line?: number;
  remediation?: string;
  detected_by: string;
  status: SecurityFindingStatus;
}

export interface SecurityAuditSummary {
  score: number;
  critical_count: number;
  high_count: number;
  medium_count: number;
  low_count: number;
  total_findings: number;
}

export interface SecurityAutoRejectionResult {
  triggered: boolean;
  reason?: string;
  retry_count: number;
  returned_phase5: boolean;
  findings?: string[];
}

export interface SecurityAuditReport {
  generated_at: string;
  summary: SecurityAuditSummary;
  status: string;
  findings: SecurityFinding[];
  dependencies?: SecurityFinding[];
  static_analysis?: SecurityFinding[];
  auto_rejection: SecurityAutoRejectionResult;
}

export interface RunSecurityAuditPayload {
  backend_dir: string;
  frontend_dir: string;
  project_root_dir: string;
  high_retry_count: number;
}
